test/tmisc.c

```diff
@@ -3978,7 +3978,7 @@ test_misc22(void)
                 if ((prec[j] + offsets[k]) > (H5Tget_size(idts[i]) * 8))
                     continue;
 
-                MESSAGE(5, ("  Testing datatypes size=%d precision=%u offset=%d\n", H5Tget_size(idts[i]),
```

Comment:
H5Tget_size returns a size_t so that specifier should be %zu here. Then, since the offsets entry is cast to an unsigned, that specifier should either be %u or the cast to unsigned should be removed and the correct specifier should be used.

Reply:
The compiler suggested %ld, but I can change it.